```
__________ KolibriTagNavigationTestCase.test_redirect_to_setup_wizard __________
self = <kolibri.core.test.test_key_urls.KolibriTagNavigationTestCase testMethod=test_redirect_to_setup_wizard>
    def test_redirect_to_setup_wizard(self):
        response = self.client.get("/")
>       self.assertEqual(response.status_code, 302)
E       AssertionError: 200 != 302
kolibri/core/test/test_key_urls.py:16: AssertionError
```
Need to update the test!

--- a/kolibri/core/test/test_key_urls.py
+++ b/kolibri/core/test/test_key_urls.py
@@ -17,7 +17,12 @@ class KolibriTagNavigationTestCase(APITestCase):
         response = self.client.get("/")
         self.assertEqual(response.status_code, 200)
         url = reverse('kolibri:setupwizardplugin:setupwizard')
-        self.assertTrue('<meta http-equiv="refresh" content="0;URL=\'{url}\'" />'.format(url=url) in response.content)
+        try:
+            content = str(response.content, 'utf-8')
+        except TypeError:
+            # Will throw TypeError on Py2 as str does not take additional argument
+            content = response.content
+        self.assertTrue('<meta http-equiv="refresh" content="0;URL=\'{url}\'" />'.format(url=url) in content)
 
     def test_redirect_root_to_user_if_not_logged_in(self):
         provision_device()

--- a/kolibri/core/test/test_key_urls.py
+++ b/kolibri/core/test/test_key_urls.py
@@ -1,11 +1,13 @@
-from __future__ import absolute_import, print_function, unicode_literals
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
 
 from django.core.urlresolvers import reverse
 from rest_framework.test import APITestCase
 
+from kolibri.auth.test.helpers import create_superuser
+from kolibri.auth.test.helpers import provision_device
 from kolibri.auth.test.test_api import FacilityFactory
-
-from kolibri.auth.test.helpers import create_superuser, provision_device
 
 DUMMY_PASSWORD = "password"
 
@@ -13,8 +15,9 @@ class KolibriTagNavigationTestCase(APITestCase):
 
     def test_redirect_to_setup_wizard(self):
         response = self.client.get("/")
-        self.assertEqual(response.status_code, 302)
-        self.assertEqual(response.get("location"), reverse('kolibri:setupwizardplugin:setupwizard'))
+        self.assertEqual(response.status_code, 200)
+        url = reverse('kolibri:setupwizardplugin:setupwizard')
+        self.assertTrue('<meta http-equiv="refresh" content="0;URL=\'{url}\'" />'.format(url=url) in response.content)
 
     def test_redirect_root_to_user_if_not_logged_in(self):
         provision_device()

--- a/kolibri/plugins/setup_wizard/middleware.py
+++ b/kolibri/plugins/setup_wizard/middleware.py
@@ -14,6 +14,8 @@ ALLOWED_PATH_LIST = [
 
 SETUP_WIZARD_URL = reverse("kolibri:setupwizardplugin:setupwizard")
 
+# Workaround for https://github.com/learningequality/kolibri/issues/3852
+# Uses a meta tag redirect rather than a 302 to prevent extra load by Windows GUI polling
 redirect_page_content = """
 <html xmlns="http://www.w3.org/1999/xhtml">
   <head>

